engine/lib/phx/src/render/draw.rs

```diff
@@ -1,486 +1,460 @@
-use super::gl;
+use std::sync::{Mutex, MutexGuard, OnceLock};
+
 use super::*;
-use crate::logging::warn;
+use crate::logging::*;
 use crate::math::*;
 use crate::system::*;
 
-/* TODO JP : Replace all immediates with static VBO/IBOs & glDraw*. */
+pub struct Draw {
+    alpha_stack: Vec<f32>,
+    color: Color,
 
-const MAX_STACK_DEPTH: usize = 16;
+    pb: PrimitiveBuilder,
+}
 
-static mut alphaStack: [f32; MAX_STACK_DEPTH] = [0.; MAX_STACK_DEPTH];
-static mut alphaIndex: i32 = -1;
-static mut color: Color = Color::WHITE;
+impl Draw {
+    fn inst() -> MutexGuard<'static, Draw> {
+        static INST: OnceLock<Mutex<Draw>> = OnceLock::new();
```

Comment:
Yeah. I want `Draw` to go away completely to be honest, but it will take a while. My plan is to introduce a new module called `Renderer` and start moving things to there.

Reply:
Also note that not every draw call goes through the Draw object. Only things like the GUI.